worlds/checksmate/__init__.py

```diff
@@ -0,0 +1,386 @@
+    piece_types_by_army = {
+        # Vanilla
+        0: {"Progressive Minor Piece": 2, "Progressive Major Piece": 1, "Progressive Major To Queen": 1},
+        # Colorbound Clobberers (the War Elephant is rather powerful)
+        1: {"Progressive Minor Piece": 1, "Progressive Major Piece": 2, "Progressive Major To Queen": 1},
+        # Remarkable Rookies
+        2: {"Progressive Minor Piece": 2, "Progressive Major Piece": 1, "Progressive Major To Queen": 1},
+        # Nutty Knights (although the Short Rook and Half Duck swap potency)
+        3: {"Progressive Minor Piece": 2, "Progressive Major Piece": 1, "Progressive Major To Queen": 1},
+        # Eurasian pieces
+        4: {"Progressive Minor Piece": 2, "Progressive Major Piece": 1, "Progressive Major To Queen": 1},
+    }
+
+    def __init__(self, multiworld: MultiWorld, player: int):
+        super(CMWorld, self).__init__(multiworld, player)
+        self.locked_locations = []
+
+    def generate_early(self) -> None:
+        army_constraint = get_option_value(self.multiworld, self.player, "fairy_chess_army")
+        if army_constraint != 0:
+            which_pieces = get_option_value(self.multiworld, self.player, "fairy_chess_pieces")
+            # Full: Adds the 12 Chess With Different Armies pieces, the Cannon, and the Vao.
+            if which_pieces == 1:
+                army_options = [0, 1, 2, 3, 4]
+            # CwDA: Adds the pieces from Ralph Betza's 12 Chess With Different Armies.
+            elif which_pieces == 2:
+                army_options = [0, 1, 2, 3]
+            # Cannon: Adds a Rook-like piece, which captures a distal chessman by leaping over an intervening chessman.
+            # Eurasian: Adds the Cannon and the Vao, a Bishop-like Cannon, in that it moves and captures diagonally.
+            elif which_pieces > 2:
+                army_options = [0, 4]
+            # Vanilla: Disables fairy chess pieces completely.
+            else:
+                army_options = [0]
+            self.army[self.player] = self.multiworld.per_slot_randoms[self.player].choice(army_options)
+        is_single = get_option_value(self.multiworld, self.player, "goal") == 0
+        if not is_single:
+            return
+        for enemy_pawn in self.item_name_groups["Enemy Pawn"]:
+            self.multiworld.start_inventory[self.player].value[enemy_pawn] = 1
+        for enemy_piece in self.item_name_groups["Enemy Piece"]:
+            self.multiworld.start_inventory[self.player].value[enemy_piece] = 1
+
+    def setting(self, name: str):
+        return getattr(self.multiworld, name)[self.player]
+
+    def fill_slot_data(self) -> dict:
+        cursed_knowledge = {name: self.multiworld.per_slot_randoms[self.player].getrandbits(31) for name in [
+            "pocket_seed", "pawn_seed", "minor_seed", "major_seed", "queen_seed"]}
+        potential_pockets = [0,0,0,0,1,1,1,1,2,2,2,2]
+        self.multiworld.per_slot_randoms[self.player].shuffle(potential_pockets)
```

Comment:
```suggestion
        self.random.shuffle(potential_pockets)
```

Reply:
Thanks, I've reflected this in my latest version (not captured).